### ISSUE TITLE
tqdm no longer works on aws lambda

### ISSUE THREAD

**Issue by eyaler** [tqdm/tqdm#466]
last working version was 4.17.1
when upgrading to 4.18.0 or 4.19.2 i get an error from aws lambda:
module initialization error: [Errno 38] Function not implemented
i am guessing this is related to multiprocessing

**Comment by casperdcl** [comment]
couldn't you post a traceback?

**Comment by eyaler** [comment]
unfortunately aws lambda does not provide a traceback. and i could not find any way to get it - if you know how please let me know.

**Comment by casperdcl** [comment]
does disabling locks using

https://github.com/tqdm/tqdm/issues/461#issuecomment-334343230

work?

**Comment by eyaler** [comment]
the: from tqdm import tqdm
triggers my error

**Comment by chengs** [comment]
@casperdcl I got some trace info from AWS lambda, as you said, it is caused by locks.

```
{
  "errorMessage": "[Errno 38] Function not implemented",
  "errorType": "OSError",
  "stackTrace": [
    [
      "/var/task/lambda_function.py",
      8,
      "lambda_handler",
      "from tqdm import tqdm"
    ],
    [
      "/var/task/tqdm/__init__.py",
      1,
      "<module>",
      "from ._tqdm import tqdm"
    ],
    [
      "/var/task/tqdm/_tqdm.py",
      53,
      "<module>",
      "mp_lock = mp.Lock()  # multiprocessing lock"
    ],
    [
      "/var/lang/lib/python3.6/multiprocessing/context.py",
      67,
      "Lock",
      "return Lock(ctx=self.get_context())"
    ],
    [
      "/var/lang/lib/python3.6/multiprocessing/synchronize.py",
      163,
      "__init__",
      "SemLock.__init__(self, SEMAPHORE, 1, 1, ctx=ctx)"
    ],
    [
      "/var/lang/lib/python3.6/multiprocessing/synchronize.py",
      60,
      "__init__",
      "unlink_now)"
    ]
  ]
}
```

**Comment by chengs** [comment]
I changed minor lines to make a fix. See https://github.com/tqdm/tqdm/pull/468

**Comment by eyaler** [comment]
@chengs  thanks! could you please share how did you get the trace stack?

**Comment by chengs** [comment]
@eyaler it's simple. 

Just need to toggle "Enable active tracking" and cause your error in the **handler function**.
![image](https://user-images.githubusercontent.com/824529/31452285-135b53c4-aee1-11e7-90e7-9dd56fddcdf9.png)

**Comment by casperdcl** [comment]
fixed in #468

**Comment by lsorber** [comment]
Any news on this issue?

**Comment by casperdcl** [comment]
Is this now fixed with `tqdm>=4.32.2`?

**Comment by casperdcl** [comment]
closing as issue seems fixed now
